lua_modules/ds18b20/ds18b20.lua

```diff
@@ -59,73 +104,119 @@ function addrs()
   return tbl
 end
 
+function convertTemperature(data, addr, unit)
+  local t = (data:byte(1) + data:byte(2) * 256)
+  if (t > 32767) then
+    t = t - 65536
+  end
+
+  if (addr:byte(1) == 0x28) then
+    t = t * 625  -- DS18B20, 4 fractional bits
+  else
+    t = 1000 * (bit.rshift(t, 1))
+    t = t - 250
+    local h = 1000 * (data:byte(8) - data:byte(7))
+    h = h / data:byte(8)
+    t = t + h
+    t = t * 10
+  end
+
+  if(unit == nil or unit == 'C') then
+   -- do nothing
+  elseif(unit == 'F') then
+    t = t * 1.8 + 320000
+  elseif(unit == 'K') then
+    t = t + 2731500
+  else
+    return "Unknown unit"
+  end
+  t = t / 10000
+  return t
+end
+
+function readScratchpad(addr)
+  ow.read(pin)
+  present = ow.reset(pin)
+  ow.select(pin, addr)
+  ow.write(pin,0xBE,1)
+  -- print("P="..present)
+  local data = nil
+  data = string.char(ow.read(pin))
+  for i = 1, 8 do
+    data = data .. string.char(ow.read(pin))
+  end
+  -- print(data:byte(1,9))
+  return data
+end
+
+function checkAddrCrc(addr)
+  crc = ow.crc8(string.sub(addr,1,7))
+  return crc == addr:byte(8)
```

Comment:
I don't think we have strict conventions but I'd write `return ow.crc8(string.sub(addr,1,7)) == addr:byte(8)` instead.